### PR TITLE
Add http basic authorization option to http requests (related to #283)

### PR DESCRIPTION
**Pull request by marisn**
This PR adds capability to access password protected WMS sources.

### CODE REVIEW

File: lib/http.c [hunk 1]
```diff
@@ -414,8 +415,41 @@ mapcache_http* mapcache_http_configuration_parse_xml(mapcache_context *ctx, ezxm
...
+        strcmp(ezxml_attr(http_node, "scheme"), "basic") == 0) {
+      ezxml_t user_node, pass_node;
+      const char *credentials;
+      user_node = ezxml_child(http_node, "user");
+      pass_node = ezxml_child(http_node, "pass");
+      if (!user_node || !pass_node) {
+        ctx->set_error(ctx, 400,
+                       "both <http> <auth> username (<user>) and pasword "
+                       "(<pass>) must be provided");
+        return NULL;
```
Comment (jratike80) [inline lib/http.c:431]: How would you provide an empty password? We at the National Land Survey of Finland have some services that use api-key as `<user>` but `<password>` is not required. At least some clients can use those services with syntax `user:` (username-colon-nothing).

**Comment by marisn** [comment]
Apparently apr_encode is available only starting from APR 1.7.0. Needs cmake checks.

File: lib/http.c (hunk 1 above)
Comment (marisn) [reply lib/http.c:431]: I had not met such services yet. I'll adjust the code.

Comment (marisn) [reply lib/http.c:431]: I checked the code. No changes are required as only the presence of xml element is performed. Thus this just will have to be documented that user or pass elements are encoded verbatim and can be left empty (`<pass></pass>`) if not required.

**Comment by marisn** [comment]
> Sorry, but I am not competent for reviewing the code. I am just a user with some experience on Mapserver.

Thanks for valuable feedback anyway. I fixed the typo and checked that your use case should work. Please mark conversation as resolved. No need to approve the PR.

**Review by tbonfort — APPROVED** [review]

**Comment by jmckenna** [comment]
thanks @marisn !

**Pull request merged**